### ISSUE TITLE
Revisit use cases related to the visualisation

### ISSUE THREAD

**Issue by GoogleCodeExporter** [joemasterjohn/debugvisualisation#43]
```
I appreciate the work done for issue 4, but (after some very basic
experiments) it is simply not enough.

As I stated before, and others have also encountered (see issue 41), that
the Local context node is disturbing.

On the other hand, by making another node as the single root (as the
introduced dig in functionality does) won't help, as it just moves the
problem, but does not solve it (e.g. by using the dig in functionality some
nodes can become hidden I'm interested in). On the other hand a forest of
disconnected graphs can be the right visualisation sometimes (as
davidmichaelcarr stated in issue 41).

I think, creating an artificial tree structure in the debug context (where
the variables and their connection does not form a tree but a general
graph) cannot succeed.

I suggest another approach handling this problem, based on both existing
solutions and new ideas:
  * The local context node should be hidden. Everytime. If the information
is needed that a node is available from the context, it could be presented
by color or something like that.
  * If the visualisation is started the same way as it is started now, the
distinct variables should become nodes, and their connections should be
only visualised. The local context variable might be used as the root, but
I'm not sure, it is a good idea.
  * It shall be possible both to clear the graph (remove every visible
item) and add single variables. This allows the user to customize the
result if needed without adding unnecessary clutter (when adding entire
trees, in most cases extra nodes are added that are not needed; and filters
can't solve this problem for every possible case).
  * If there is no typical use for the dig in (I can't imagine any), that
it shall be removed, as it is hard to imagine, how it works.

In short: the user interface should not work like a visualised tree
structure with extra arcs, but as general graphs (maybe not even
connected). The Zest framework is capable of visualising that.

PS. sorry, if I was a bit too harsh. And yes, I think, this is high
priority, as this problem undermines the usability of the plugin.
```

Original issue reported on code.google.com by `Stampie.hun@gmail.com` on 16 Sep 2009 at 8:47
- Blocking: #41
- Blocked on: #29, #54
